verifier/hash_verifier.go

```diff
@@ -307,7 +314,29 @@ func (v *Verifier) compareHashAndUpdate(challengeId uint64, chainRootHash []byte
 		return err
 	}
 	// update metrics if no err
+	logging.Logger.Infof("challenge succeeded for challengeId: %d, hash mismatched", challengeId)
 	v.metricService.IncVerifiedChallenges()
 	v.metricService.IncChallengeSuccess()
 	return err
 }
+
+func (v *Verifier) isInternalSP(spEndpoint string) bool {
+	for _, internalEndpoint := range v.config.SPConfig.InternalSPEndpoints {
+		if strings.Contains(spEndpoint, internalEndpoint) {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Verifier) updateDeduplicationIntervalLoop() {
+	ticker := time.NewTicker(UpdateDeduplicationInterval)
+	for range ticker.C {
+		updatedDeduplicationInterval, err := v.executor.QueryChallengeSlashCoolingOffPeriod()
+		if err != nil {
+			logging.Logger.Errorf("error updating deduplication interval, err=%s", err.Error())
+			return
+		}
+		v.deduplicationInterval = updatedDeduplicationInterval
```

Comment:
We may need a mutex or lock here.

Reply:
and where `v.deduplicationInterval` been read.

Reply:
fixed